src/rules/dot_product/in1.jl

```diff
@@ -4,5 +4,40 @@ export rule
     x  = mean(m_in2)
     xi = x * weightedmean(m_out)
     w  = correction!(meta, x * precision(m_out) * x')
-    return MvGaussianWeightedMeanPrecision(xi, w)
+    return MvNormalWeightedMeanPrecision(xi, w)
+end
+
+@rule typeof(dot)(:in1, Marginalisation) (m_out::UnivariateNormalDistributionsFamily, m_in2::PointMass, meta::AbstractCorrection) = begin
+    x  = mean(m_in2)
+    xi = x * weightedmean(m_out)
+    w  = correction!(meta, x * precision(m_out) * x')
+    return NormalWeightedMeanPrecision(xi, w)
+end
+
+@rule typeof(dot)(:in1, Marginalisation) (m_out::UnivariateNormalDistributionsFamily, m_in2::PointMass) = begin
+    x  = mean(m_in2)
+    xi = x * weightedmean(m_out)
+    w  = x * precision(m_out) * x'
+    return NormalWeightedMeanPrecision(xi, w)
+end
+
+@rule typeof(dot)(:in1, Marginalisation) (m_out::PointMass{ <: AbstractVector }, m_in2::UnivariateNormalDistributionsFamily, meta::AbstractCorrection) = begin
+    x  = mean(m_out)
+    xi = x * weightedmean(m_in2)
+    w  = correction!(meta, x * precision(m_in2) * x')
```

Comment:
Please combine `weighedmean(m_in2)` and `precision(m_in2)` into `weightedmean_precision(m_in2)` for possibly improved performance, also for the other rules.